src/sentry/api/utils.py

```diff
@@ -170,15 +225,15 @@ def generate_organization_hostname(org_slug: str) -> str:
 
 
 def generate_organization_url(org_slug: str) -> str:
-    org_url_template = options.get("system.organization-url-template")
+    org_url_template: str = options.get("system.organization-url-template")
     if not org_url_template:
-        return options.get("system.url-prefix")
+        return options.get("system.url-prefix")  # type: ignore[no-any-return]
     return org_url_template.replace("{hostname}", generate_organization_hostname(org_slug))
 
 
 def generate_region_url() -> str:
-    region_url_template = options.get("system.region-api-url-template")
+    region_url_template: str = options.get("system.region-api-url-template")
     region = options.get("system.region") or None
     if not region_url_template or not region:
-        return options.get("system.url-prefix")
+        return options.get("system.url-prefix")  # type: ignore[no-any-return]
```

Comment:
I suppose there's no way to bound this type any tighter than Any?
Any effectively shuts off type checking for anything/everything above this function in the stack.

Even this nasty thing would be better:
```
Jsonish = Union[ None, bool, int, str, float, list["Jsonish"], dict[str, "Jsonish"]
```

P4

Reply:
mypy has rudimentary cyclical type support but we haven't turned it on yet because it breaks a bunch of stuff -- for return values it's also not helpful since you'd then have to cast away the union anyway

Reply:
also your assumption about shutting off type checking is incorrect -- this function has `-> str` so callers will see that.  the unsoundness is limited to this one return statement